### PR TITLE
Add Gmock for UefiBootServicesTableLib

### PR DESCRIPTION
**Pull request by VivianNK**
## Description

Create Gmock for UefiBootServicesTableLib, with a portion of the mock function declarations.

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Consumed the mock in a local GoogleTest.

## Integration Instructions
N/A

### CODE REVIEW

**Comment by apop5** [comment]
I think this needs added to MdeModulePkgHostTest.dsc as well so that it is compiled as part of CI testing.

**Review by apop5 — APPROVED** [review]

File: MdePkg/Test/Mock/Library/GoogleTest/MockUefiBootServicesTableLib/MockUefiBootServicesTableLib.inf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+[Packages]
+  MdePkg/MdePkg.dec
+  UnitTestFrameworkPkg/UnitTestFrameworkPkg.dec
+
+[LibraryClasses]
+  GoogleTestLib
+
+[BuildOptions]
+  MSFT:*_*_*_CC_FLAGS = /EHsc
```
Comment (kuqin12) [inline MdePkg/Test/Mock/Library/GoogleTest/MockUefiBootServicesTableLib/MockUefiBootServicesTableLib.inf:33]: Can you please remind me what this flag was for?

**Review by kuqin12 — APPROVED** [review]

File: MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiBootServicesTableLib.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    gBS_CloseEvent,
+    (IN EFI_EVENT Event)
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    gBS_HandleProtocol,
+    (IN  EFI_HANDLE               Handle,
```
Comment (Flickdm) [inline MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiBootServicesTableLib.h]: NIT: what's going on with the indent?

**Review by Flickdm — APPROVED** [review]

File: MdePkg/Test/Mock/Library/GoogleTest/MockUefiBootServicesTableLib/MockUefiBootServicesTableLib.inf (hunk 1 above)
Comment (VivianNK) [reply MdePkg/Test/Mock/Library/GoogleTest/MockUefiBootServicesTableLib/MockUefiBootServicesTableLib.inf:33]: Exception Handling flag for C++.
This enables C++ stack unwinding and assumes functions declared as extern"C" never throw a C++ exception.

https://learn.microsoft.com/en-us/cpp/build/reference/eh-exception-handling-model?view=msvc-170

**Pull request merged**